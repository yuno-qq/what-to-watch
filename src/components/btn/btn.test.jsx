import React from 'react'
import renderer from 'react-test-renderer'
import {describe, test, expect} from "@jest/globals"

import Btn from "./btn.jsx"


describe(`<Btn> snapshots:`, () => {
  test(`should render basic case`, () => {
    const btn = renderer.create(<Btn renderIcon={() => <svg></svg>} title={`Title of button`}/>)
      .toJSON()

    expect(btn).toMatchSnapshot()
  })

  test(`should render with all props`, () => {
    const btn = renderer.create(<Btn renderIcon={() => <svg></svg>}
      title={`Title of 2nd button`}
      cssMod={`btn--size_sm`}
      cssMix={`player__btn`}
      onClick={() => {}}
    />).toJSON()

    expect(btn).toMatchSnapshot()
  })
})
