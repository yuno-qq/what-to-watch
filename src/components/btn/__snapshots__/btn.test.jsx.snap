// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Btn> snapshots: should render basic case 1`] = `
<button
  className="btn  "
  onClick={[Function]}
  type="button"
>
  <svg />
  <span>
    Title of button
  </span>
</button>
`;

exports[`<Btn> snapshots: should render with all props 1`] = `
<button
  className="btn btn--size_sm player__btn"
  onClick={[Function]}
  type="button"
>
  <svg />
  <span>
    Title of 2nd button
  </span>
</button>
`;
